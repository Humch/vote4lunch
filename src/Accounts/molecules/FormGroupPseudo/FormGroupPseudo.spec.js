/* eslint-disable no-undef */
import React from 'react';
import { Provider } from 'react-redux';

import { mount } from 'enzyme';

import renderer from 'react-test-renderer';

import configureStore from 'redux-mock-store';

import FormGoupPseudo from './FormGroupPseudo';

describe('FormGoupPseudo Snapshot', () => {
  const initialState = {
    userData: {
      pseudo: '',
    },
    pseudoChecker: {
      payload: {
        exist: false,
      },
    },
  };
  const text = 'le super label';
  const forProp = 'pseudo';
  const mockStore = configureStore();
  let store;

  beforeEach(() => {
    store = mockStore(initialState);
  });

  it('capturing Snapshot of FormGoupPseudo', () => {
    const renderedValue = renderer.create(
      <Provider store={store}>
        <FormGoupPseudo
          text={text}
          forProp={forProp}
        />
      </Provider>
    ).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});

describe('FormGoupPseudo Snapshot', () => {
  const initialState = {
    userData: {
      pseudo: 'bob',
    },
    pseudoChecker: {
      payload: { exist: false },
      loading: false,
    }
  };
  const text = 'le super label';
  const forProp = 'pseudo';
  const mockStore = configureStore();
  let store;

  beforeEach(() => {
    store = mockStore(initialState);
  });

  it('capturing Snapshot of FormGoupPseudo with pseudo', () => {
    const renderedValue = renderer.create(
      <Provider store={store}>
        <FormGoupPseudo
          text={text}
          forProp={forProp}
        />
      </Provider>
    ).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});

describe('FormGoupPseudo - REACT-REDUX (Mount + wrapping in <Provider>)', () => {
  const initialState = {
    userData: {
      pseudo: '',
    },
    pseudoChecker: {
      payload: { exist: false },
      loading: false,
    }
  };
  const text = 'le super label';
  const forProp = 'pseudo';
  const mockStore = configureStore();
  let store;
  let wrapper;

  beforeEach(() => {
    store = mockStore(initialState);
  });

  it('render the connected(SMART) component', () => {
    wrapper = mount(
      <Provider store={store}>
        <FormGoupPseudo
          text={text}
          forProp={forProp}
        />
      </Provider>
    );
    expect(wrapper.length).toEqual(1);
  });

  it('expect the input have is-invalid className', () => {
    initialState.pseudoChecker.payload.exist = true;
    wrapper = mount(
      <Provider store={store}>
        <FormGoupPseudo
          text={text}
          forProp={forProp}
        />
      </Provider>
    );
    const input = wrapper.find('input');

    expect(input.hasClass('is-invalid')).toBe(true);
  });
});
