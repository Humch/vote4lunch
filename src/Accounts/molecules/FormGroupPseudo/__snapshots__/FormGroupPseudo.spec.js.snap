// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormGoupPseudo Snapshot capturing Snapshot of FormGoupPseudo 1`] = `
<div
  className="form-group"
>
  <label
    className=""
    htmlFor="pseudo"
  >
    le super label
  </label>
  <div
    className="input-group"
  >
    <input
      className="form-control "
      id="pseudo"
      name="pseudo"
      onChange={[Function]}
      required={true}
      type="text"
      value=""
    />
    
    <div
      className="invalid-feedback"
    >
      Ce pseudo existe déjà
    </div>
  </div>
</div>
`;

exports[`FormGoupPseudo Snapshot capturing Snapshot of FormGoupPseudo with pseudo 1`] = `
<div
  className="form-group"
>
  <label
    className=""
    htmlFor="pseudo"
  >
    le super label
  </label>
  <div
    className="input-group"
  >
    <input
      className="form-control "
      id="pseudo"
      name="pseudo"
      onChange={[Function]}
      required={true}
      type="text"
      value="bob"
    />
    <div
      className="input-group-append"
    >
      <span
        className="input-group-text"
      >
        <span>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-check fa-w-16 "
            color="green"
            data-icon="check"
            data-prefix="fas"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </span>
      </span>
    </div>
    <div
      className="invalid-feedback"
    >
      Ce pseudo existe déjà
    </div>
  </div>
</div>
`;
